boot/mockboot.go

```diff
@@ -0,0 +1,38 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// This file includes helper functions that can be used by other
+// packages unit tests.
+
+package boot
+
+import (
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/secboot/keys"
+)
+
+type SealKeyToModeenvFlags = sealKeyToModeenvFlags
+
+func MockSealKeyToModeenv(f func(key, saveKey keys.EncryptionKey, model *asserts.Model, modeenv *Modeenv, flags SealKeyToModeenvFlags) error) (restore func()) {
```

Comment:
func MockResealKeyToModeenv is actually in seal.go, so this probably should go there. We try not to have separate mock dedicated source files